cmip6_downscaling/methods/bcsd.py

```diff
@@ -65,54 +83,119 @@ def make_flow_paths(
     return coarse_obs_path, spatial_anomalies_path, bias_corrected_path, final_out_path
 
 
-def return_obs(train_period_start: str, train_period_end: str, variable: str) -> xr.Dataset:
+def return_obs(
+    gcm: str,
+    scenario: str,
+    train_period_start: str,
+    train_period_end: str,
+    predict_period_start: str,
+    predict_period_end: str,
+    variable: str,
+    latmin: str,
+    latmax: str,
+    lonmin: str,
+    lonmax: str,
+) -> xr.Dataset:
     """Loads ERA5 observation data for given time bounds and variable
 
     Parameters
     ----------
+    gcm : str
+        Input GCM
+    scenario: str
+        Input GCM scenario
     train_period_start : str
-        Starting time bounds
+        Date for training period start (e.g. '1985')
     train_period_end : str
-        Ending time bounds
-    variable : str
-        ERA5 variable. ex: 'tasmax'
+        Date for training period end (e.g. '2015')
+    predict_period_start : str
+        Date for prediction period start (e.g. '2090')
+    predict_period_end : str
+        Date for prediction period end (e.g. '2090')
+    variable: str
+        The variable included in the dataset.
+    latmin : str
+            From run hyperparameters
+    latmax : str
+            From run hyperparameters
+    lonmin : str
+            From run hyperparameters
+    lonmax : str
+            From run hyperparameters
 
     Returns
     -------
     xr.Dataset
         Loaded xarray dataset of ERA5 observation data. Chunked in time: 365
     """
-    obs_ds = open_era5(variable, start_year=train_period_start, end_year=train_period_end)
+    obs_load = open_era5(variable, start_year=train_period_start, end_year=train_period_end)
+    obs_ds = subset_dataset(
+        obs_load, train_period_start, train_period_end, latmin, latmax, lonmin, lonmax
+    )
     # Chunking the observation dataset by 'time':365 fixes irregular zarr chunking issues caused by leap-years.
     obs_ds = obs_ds.chunk({'time': 365})
     return obs_ds
 
 
-def get_coarse_obs(obs_ds: xr.Dataset, variable: str) -> xr.Dataset:
+def get_coarse_obs(
+    obs_ds: xr.Dataset,
+    gcm: str,
+    scenario: str,
+    train_period_start: str,
+    train_period_end: str,
+    predict_period_start: str,
```

Comment:
`predict_period` variables shouldn't be required here

Reply:
Currently for the caching string, I have to pass all the params used for naming in. I think once we move this to Cindy's naming funcs, this can be updated.